### PR TITLE
feat(context): add the ability to create and delete a compute context

### PR DESCRIPTION
**Pull request by krishna-acondy**
## `createContext`
* Create a compute context with the supplied parameters.
* Fail fast if one of the parameters is missing.

## `deleteContext`
* Delete a context with the given ID.
* Fail fast if the ID is missing.

Enables https://github.com/sasjs/cli/issues/58 to be implemented.

### CODE REVIEW

**Review by medjedovicm — DISMISSED** [review]

File: src/SASjs.ts [hunk 1]
```diff
@@ -107,6 +107,34 @@ export default class SASjs {
     return await this.sasViyaApiClient!.getExecutableContexts(accessToken)
   }
 
+  public async createContext(
```
Comment (YuryShkoda) [inline src/SASjs.ts:110]: all public functions need to be documented

File: src/SASjs.ts [hunk 2]
```diff
@@ -107,6 +107,34 @@ export default class SASjs {
...
+      contextName,
+      launchContextName,
+      sharedAccountId,
+      autoExecLines,
+      authorizedUsers,
+      accessToken
+    )
+  }
+
+  public async deleteContext(contextId: string, accessToken?: string) {
```
Comment (YuryShkoda) [inline src/SASjs.ts:131]: all public functions need to be documented

File: src/SASViyaApiClient.ts [hunk 3]
```diff
@@ -193,6 +193,109 @@ export class SASViyaApiClient {
...
+      requestBody['authorizedUsers'] = authorizedUsers
+    } else {
+      requestBody['authorizeAllAuthenticatedUsers'] = true
+    }
+
+    if (autoExecLines) {
+      requestBody.environment = { autoExecLines }
+    }
+
+    console.log('Body', requestBody)
```
[the remark this answers is not in the document]
Comment (krishna-acondy) [reply src/SASViyaApiClient.ts]: Whoops, that's gone now.

File: src/SASjs.ts (hunk 1 above)
Comment (krishna-acondy) [reply src/SASjs.ts:119]: I've added comments for the new functions I added - but if you look closely, none of the other public functions in this file are documented either.
We'll need to address that as part of a separate issue.

File: src/SASjs.ts (hunk 2 above)
Comment (krishna-acondy) [reply src/SASjs.ts:145]: I've added comments for the new functions I added - but if you look closely, none of the other public functions in this file are documented either.
We'll need to address that as part of a separate issue.

Comment (YuryShkoda) [reply src/SASjs.ts:145]: Yes, you are right that the majority of the functions are not documented and we should create a separate issue. But still, it'll be easier to add comments as we go.

**Review by YuryShkoda — APPROVED** [review]

**Pull request merged**